src/http.ts

```diff
@@ -250,19 +258,23 @@ function httpRequest<T>(url: string, options?: RequestInit, errorTransform: any
 					// Error parsing response, so we'll just use what we have
 					error = { ...new Error(), ...convertResponse(response) };
 				}
-				throw errorTransform !== null ? errorTransform(error) : error;
+				throw error;
 			}
 			const text = await response.text();
 			const isJson = !!text && (text[0] === '{' || text[0] === '[');
-			let data: T;
+			let data: TData;
 			try {
 				// eslint-disable-next-line @typescript-eslint/no-explicit-any
-				data = (isJson ? JSON.parse(text) : (text as any)) as T;
+				data = (isJson ? JSON.parse(text) : (text as any)) as TData;
 			} catch {
 				// eslint-disable-next-line @typescript-eslint/no-explicit-any
-				data = (text as any) as T;
+				data = (text as any) as TData;
 			}
 			return convertResponse(response, data);
+		}),
+		catchError(error => {
+			const transformed = errorTransform?.(error) ?? error;
+			return throwError(transformed);
```

Comment:
Does this capture the intent ... transforming the error before throwing it?

Reply:
Yes, that provides us the transformed error messaging to present in the UI instead of the not so understandable message coming from the API.